File: Version_Java/Medidor.java
package Version_Java;

public class Medidor {

    public static Resultado medirLinhas(int[][] m) {
        long inicio = System.nanoTime();
        long soma = 0;

        // percorre a matriz por linhas (jeito sigma de acessar a matriz - melhor uso de cache)
        for (int i = 0; i < m.length; i++)
            for (int j = 0; j < m.length; j++)
                soma += m[i][j];

        long fim = System.nanoTime();

        return new Resultado("Linhas", soma, (fim - inicio));
    }

    public static Resultado medirColunas(int[][] m) {
        long inicio = System.nanoTime();
        long soma = 0;

        // percorre a matriz por colunas (jeito burro de acessar a matriz - pior uso de cache)
        for (int i = 0; i < m.length; i++)
            for (int j = 0; j < m.length; j++)
                soma += m[j][i];

        long fim = System.nanoTime();

        return new Resultado("Colunas", soma, (fim - inicio));
    }
}

File: Version_Python/Medidor.py
import time

def medir_linhas(m):
    inicio = time.perf_counter_ns()
    soma = 0

    # jeito inteligente
    for i in range(len(m)):
        for j in range(len(m)):
            soma += m[i][j]

    fim = time.perf_counter_ns()
    return ("Linhas", soma, fim - inicio)

def medir_colunas(m):
    inicio = time.perf_counter_ns()
    soma = 0

    # jeito burro
    for i in range(len(m)):
        for j in range(len(m)):
            soma += m[j][i]

    fim = time.perf_counter_ns()
    return ("Colunas", soma, fim - inicio)